app/models/subscription.rb

```diff
@@ -5,13 +5,34 @@ class Subscription < ApplicationRecord
   has_many :subscription_contents
 
   enum frequency: { immediately: 0, daily: 1, weekly: 2 }
-  enum source: { user_signed_up: 0, frequency_changed: 1, imported: 2 }
+  enum source: { user_signed_up: 0, frequency_changed: 1, imported: 2 }, _prefix: true
+  enum ended_reason: { unsubscribed: 0, non_existant_email: 1, frequency_changed: 2 }, _prefix: :ended
 
-  validates :subscriber, uniqueness: { scope: :subscriber_list }
+  validates_uniqueness_of :subscriber, scope: :subscriber_list, conditions: -> { active }
+  validate :both_ended_at_and_ended_reason_set
 
-  scope :not_deleted, -> { where(ended_at: nil) }
+  scope :active, -> { where(ended_at: nil) }
 
-  def destroy
-    update_attributes!(ended_at: Time.now)
+  def ended?
+    ended_at.present? && ended_reason.present?
```

Comment:
I'd not worry about sweating about whether ended_reason is set as well - it makes it all a little more complicated (and conflicts with the scope).

Whereas I think it's simpler if we just consider presence of ended_at to mean it's ended and we just do what we can to prevent the reason being null.

Reply:
Sounds fine to me, I've amended the PR with this.